### PR TITLE
upgrade to 2.23.1 of lsquic

### PR DESCRIPTION
**Pull request by vhuynh3000**

### CODE REVIEW

**Review by saltycrys — CHANGES_REQUESTED** [review]
Please also update the version in `shard.yml`.

**Comment by Perflyst** [comment]
@vhuynh3000 can you please take a look at the review from saltycrys?

**Comment by unixfox** [comment]
> @vhuynh3000 can you please take a look at the review from saltycrys?

If @vhuynh3000 doesn't respond you or @saltycrys can update the code in the pull request and update the `shard.yml`. It shouldn't be too hard.

**Comment by Perflyst** [comment]
Right, but there is already [v2.28.0](https://github.com/litespeedtech/lsquic/releases/tag/v2.28.0), not sure if it's worth to update only to v2.23.1

**Comment by unixfox** [comment]
Well v2.28.0 will require again new code changes and at least the job is already done for v2.23.1. v2.23.1 bring new bugs fixed and that's something we welcome very much.

**Comment by Perflyst** [comment]
I guess we should use the liblsquic from https://github.com/iv-org/lsquic-static-alpine. Just opened https://github.com/iv-org/lsquic-static-alpine/pull/1

**Comment by unixfox** [comment]
Sure if you create a new tag on https://github.com/iv-org/lsquic-static-alpine I can test on my instance.

**Comment by Perflyst** [comment]
You can get the build from the PR here: https://github.com/iv-org/lsquic-static-alpine/suites/1995406313/artifacts/39816860

**Comment by saltycrys** [comment]
https://github.com/iv-org/lsquic-static-alpine/pull/1 had a wrong commit for BoringSSL, fixed in https://github.com/iv-org/lsquic-static-alpine/pull/2.

Release available here: https://github.com/iv-org/lsquic-static-alpine/releases/tag/v2.23.1
liblsquic.a v2.23.1 available here: https://github.com/iv-org/lsquic-static-alpine/releases/download/v2.23.1/liblsquic.a

**Comment by Perflyst** [comment]
Patch is now in testing on my server

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Crystal bindings to the excellent [LSQUIC](https://github.com/litespeedtech/lsquic) library.
 
-Releases track lsquic's versioning starting with `v2.18.1`.
+Releases track lsquic's versioning starting with `v2.23.1`.
```
Comment (saltycrys) [inline README.md:5]:
```suggestion
Releases track lsquic's versioning starting with `v2.18.1`.
```
This is about the version scheme changing with `v2.18.1`.

**Review by saltycrys — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Perflyst) [reply README.md:5]: ugh, I just grepped and replaced, sorry

**Comment by Perflyst** [comment]
The patch is working fine for 3 days on one of my backend, I guess it's ready. Thanks again @vhuynh3000

**Pull request merged**